### PR TITLE
Update the debugger support so it properly launches under a debugger, and supports attach to a running job

### LINKED ISSUE #3660: error launching/attaching LaunchMON debugger with OpenMPI 2.1.1
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)
v2.1.1

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)
from source tarball

### Please describe the system on which you are running

* Operating system/version: RHEL7
* Computer hardware: x86-64
* Network type: infiniband

-----------------------------

## Details of the problem
I am having trouble attaching LaunchMON when using OpenMPI 2.1.1.

```bash-4.2$ ./test.launch_1
[LMON_FE] launching the job/daemons via /usr/workspace/wsrzd/lee218/install/toss_3_x86_64_ib/ompi-2.1.1/bin/orterun

[LMON FE] 6 RM types are supported
[warn] Epoll ADD(4) on fd 38 failed.  Old events were 0; read change was 0 (none); write change was 1 (add): Bad file descriptor
[warn] Epoll ADD(4) on fd 35 failed.  Old events were 0; read change was 0 (none); write change was 1 (add): Bad file descriptor
[rzoz1:132008] [[21665,0],0] usock_peer_send_blocking: send() to socket 36 failed: Broken pipe (32)
[rzoz1:132008] [[21665,0],0] ORTE_ERROR_LOG: Unreachable in file oob_usock_connection.c at line 316
[rzoz1:132008] [[21665,0],0]-[[21665,1],1] usock_peer_accept: usock_peer_send_connect_ack failed
[rzoz1:132008] [[21665,0],0] usock_peer_send_blocking: send() to socket 40 failed: Broken pipe (32)
[rzoz1:132008] [[21665,0],0] ORTE_ERROR_LOG: Unreachable in file oob_usock_connection.c at line 316
[rzoz1:132008] [[21665,0],0]-[[21665,1],0] usock_peer_accept: usock_peer_send_connect_ack failed
--------------------------------------------------------------------------
orterun was unable to start the specified application as it encountered an
error:

Error name: Not supported
Node: rzoz1

when attempting to start process rank 0.
--------------------------------------------------------------------------
<Jun 06 11:51:50> <LMON FE API> (INFO): FE-ENGINE connection timed out: 120
[LMON FE] FAILED
```

Here's how you can reproduce (modify your PATH and the path to mpirun):

```
git clone https://github.com/llnl/launchmon.git
cd launchmon/
export PATH=/collab/usr/global/tools/openmpi/toss_3_x86_64_ib/openmpi-2.1.1/bin:$PATH
CFLAGS="-g -O0" CXXFLAGS="-g -O0" ./configure --prefix=/nfs/tmp2/lee218/prefix/launchmon-1.0.3b --with-test-rm=orte --with-test-rm-launcher=/collab/usr/global/tools/openmpi/toss_3_x86_64_ib/openmpi-2.1.1/bin/mpirun --with-test-installed --with-test-nnodes=1 && make clean && make -j 8 install && make -j 8 check
cd test/src
./test.launch_1
```

In addition, the LaunchMON "test.attach_1" test hangs when trying to attach. @rhc54 had previously helped me with various debugger attach issues and we had a working commit. I don't know if that made it into the release or if this is a new issue. It would be nice if LaunchMON tests could be integrated as part of the release testing.

### PR DESCRIPTION
**Pull request by rhc54**
Thanks to Greg Lee for the report

Fixes #3660

Signed-off-by: Ralph Castain <rhc@open-mpi.org>

### CODE REVIEW

File: orte/tools/orterun/orterun.c [hunk 1]
```diff
@@ -2414,7 +2414,7 @@ static void setup_debugger_job(void)
      */
     orte_plm_base_create_jobid(debugger);
     /* set the personality to ORTE */
```
Comment (jjhursey) [inline orte/tools/orterun/orterun.c]: Should probably update the comment to `ompi` instead of `orte` to reflect the line that follows.

**Comment by rhc54** [comment]
@jjhursey Done!

**Comment by lee218llnl** [comment]
This looks good to me, thanks!

File: orte/tools/orterun/orterun.c [hunk 2]
```diff
@@ -2413,8 +2413,8 @@ static void setup_debugger_job(void)
      * to avoid confusing the rest of the system's bookkeeping
      */
     orte_plm_base_create_jobid(debugger);
-    /* set the personality to ORTE */
-    debugger->personality = strdup("orte");
+    /* set the personality to OMPI */
+    debugger->personality = strdup("ompi");
```
Comment (jsquyres) [inline orte/tools/orterun/orterun.c]: @rhc54 Just curious -- why change this to "ompi" inside of `orterun`?

Comment (hppritcha) [inline orte/tools/orterun/orterun.c]: why the change here?

**Comment by hppritcha** [comment]
Do we need this fix in v3.0.x?

File: orte/tools/orterun/orterun.c (hunk 2 above)
Comment (rhc54) [inline orte/tools/orterun/orterun.c]: Because you folks removed the "orte" schizo component in 2.x, so we have to default to the "ompi" one 😄

**Comment by rhc54** [comment]
It's already in master and 3.x - the problem was the prior branches
